Admin overview doc (#6412)
knative/docs: docs/versioned/admin/admin-overview.md

Of particular interest to cluster administrators is that Knative supports customizable _default values_ on the parameters defined in resource YAML files. These configurations reduce the amount of environment configuration tasks developers needs to consider.

## Knative installations

See the [Installation roadmap](../install/README.md#installation-roadmap) for prerequisites and installation steps. Your first installation decision is whether to use a YAML-based installation or use the Knative Operator. The Knative Operator is a custom controller that extends the Kubernetes API to install Knative components. If you just need to get acquainted with Knative at this time, you can install the quickstart.

## Configuring Knative

> Knative uses Kubernetes-style YAML manifests to define and configure system components. These manifests include core resources, custom resource definitions (CRDs), and extensibility features. As with Kubernetes, these configuration resources are declarative and managed using the `kubectl` CLI tool.

Reviewer: ```suggestion
Knative uses Kubernetes YAML manifests to define and configure system components. These manifests include core resources, custom resource definitions (CRDs), and extensibility features. As with Kubernetes, these configuration resources are declarative and can be managed using the `kubectl` CLI tool or with continuous delivery tools.
```

- [Istio access to deployed services](../serving/istio-authorization.md)
- [Namespace exclusion from webhooks](../serving/istio-authorization.md)

Configurations for maintenance:

- [Garbage collection](../serving/revisions/revision-admin-config-options.md)
- [High availability](../serving/config-ha.md)
- [Rollout duration for revisions](../serving/configuration/rolling-out-latest-revision-configmap.md)
- [Autoscaling of Kafka features](../eventing/configuration/keda-configuration.md)

Configurations for security encryptions:

- [cert-manager](../serving/encryption/configure-certmanager-integration.md)
- [External domains](../serving/encryption/external-domain-tls.md)
- [Local domains](../serving/encryption/cluster-local-domain-tls.md)
- [system-internal](../serving/encryption/system-internal-tls.md)

Configurations for extensions:

- [Kafka Broker features](../serving/encryption/system-internal-tls.md)
- [Sugar Controller](../eventing/configuration/sugar-configuration.md)

Configurations for flagging features:

- [Serving features](../serving/configuration/feature-flags.md)
- [Eventing features](../eventing/features/README.md)

## Authorizations

> You can grant developers access to additional resources related to their namespace in other services, such as observability, logs, metrics, tracing, and dashboards.

Reviewer: Not for this PR, but we should expand this to also talk about how application access can be managed by using ingress and Eventing features to require specific types of tokens (authentication and authorization) to deliver requests or events.

## Knative installations

See the [Installation roadmap](../install/README.md#installation-roadmap) for prerequisites and installation steps. Your first installation decision is whether to use a YAML-based installation or use the Knative Operator. If you just need to get acquainted with Knative at this time, you can install the quickstart.

## Configuring Knative

Knative uses Kubernetes-style YAML manifests to define and configure system components. These manifests include core resources, custom resource definitions (CRDs), and extensibility features. As with Kubernetes, these configuration resources are declarative and managed using the `kubectl` CLI tool.

### Resource scoping and namespaces

Knative resources are associated with namespaces. Knative adheres to the Kubernetes model of namespace-based isolation, enabling you to control the following:

- Development teams and resources by assigning developers to teams defined by namespaces.

> - The capabilities applications have by referencing a namespace.

Author: Here's my fix:

Knative resources are associated with namespaces. Knative adheres to the Kubernetes model of namespace-based isolation, enabling you to manage development teams and resources by assigning developers to teams defined by namespaces. There is also management of functionality by the applications that reference a namespace.

> - Exclude webhooks from using a namespace, useful for apps in development.

Author: removed

- [Autoscaling of Kafka features](../eventing/configuration/keda-configuration.md)

Configurations for security encryptions:

- [cert-manager](../serving/encryption/configure-certmanager-integration.md)
- [External domains](../serving/encryption/external-domain-tls.md)
- [Local domains](../serving/encryption/cluster-local-domain-tls.md)
- [system-internal](../serving/encryption/system-internal-tls.md)

Configurations for extensions:

- [Kafka Broker features](../serving/encryption/system-internal-tls.md)
- [Sugar Controller](../eventing/configuration/sugar-configuration.md)

Configurations for flagging features:

- [Serving features](../serving/configuration/feature-flags.md)
- [Eventing features](../eventing/features/README.md)

## Authorizations

You can grant developers access to additional resources related to their namespace in other services, such as observability, logs, metrics, tracing, and dashboards.

## Upgrades

More often than not, you will be performing upgrades to your cluster's infrastructure and apps and services. Knative is designed and tested for continuous operation during upgrades and rollbacks, allowing you to:

> - Upgrade or revert a Knative cluster while it is serving traffic, rather than needing a maintenance window.

Reviewer: ```suggestion
- Upgrade or revert the Knative components while it is serving traffic, rather than needing a maintenance window.
```

---
audience: administrator
components:
  - serving
  - eventing
function: reference
---
# Overview

Knative has a set of tools and capabilities to administer your Kubernetes clusters. This article provides a overview of Knative features, capabilities, and resources of interest to Kubernetes Administrators, and is organized by the following areas:

- Installation
- Configuration
- Monitoring and Observability
- Security and Access Control
- Updates and Maintenance

## Installation

> Use the Installation Roadmap for guidance on installing Knative using the resources and tool choices that best fit your needs. You can install using YAML resources, manifests, or CLI tools that include the Knative Operator.

Reviewer: ```suggestion
Use the [Installation Roadmap](../install/README.md) for guidance on installing Knative using the resources and tool choices that best fit your needs. You can install using YAML manifests or a Kubernetes operator; the operator supports management via the `kn` CLI.
```

"manifests" is a term for resources declared in YAML format, so those two sort of mean the same thing.

Reviewer: It may be worth explaining that Knative can add support for new transports after the initial installation, so your initial choices don't lock you in long-term (though migrating from one message transport or network ingress to another does need to be planned to avoid losing messages).

The Knative Operator is a custom controller that extends the Kubernetes API to install Knative components.

You can install Knative components in three ways:

- Use a [YAML-based installation](/install/yaml-install/README.md)
- Use the [Knative Operator CLI plugin](/install/operator/knative-with-operator-cli.md)
- Use the [Knative Operator](/install/operator/knative-with-operators.md) with YAML resource files and the `kubectl` CLI.

    - [Knative Serving installation files](/install/yaml-install/serving/serving-installation-files.md)

>     - [Knative Eventing installation files](/install/yaml-install/eventing/eventing-installation-files.md)

Reviewer: I don't know if you need these links given the installation overview, but if you do, you should put them under the first bullet point.

### Recommended plugins

You can also install these plugins service to extend Knative capabilities for service meshes and application security:

- [Istio for Knative](/install/installing-istio.md)

> - [Knative Backstage plugin](/install/installing-backstage-plugins.md)

Reviewer: The backstage plugin isn't a security tool, but it can help developers understand what's available in their cluster by providing a dashboard.

I would specifically call out `cert-manager` as another component which is optional but highly recommended.

> - [Installing Security-Guard](/serving/app-security/security-guard-install.md)

Reviewer: Security Guard is a bit of an experiment; I wouldn't necessarily recommend it as part of a core installation.

For un-installations, see [Uninstalling Knative](/install/uninstall.md).

## Configurations

> Knative provides optimization capabilities for the Serving and Eventing components and settings for administering your clusters.

Reviewer 2: ```suggestion
Knative enables you to  optimize Serving and Eventing components and configure administration settings for your clusters.
```

### ConfigMaps

The Knative Operator propagates values from custom resources to the ConfigMaps object, a storage object of non-confidential values in key-value pairs. You can [Configuring Knative using the Operator](/install/operator/configuring-with-operator.md) and define custom resource definitions:

- [Knative Serving CRDs](/install/operator/configuring-serving-cr.md)

> - [Knative Eventing CRDs](/install/operator/configuring-eventing-cr.md)

Reviewer 2: Provide descriptions in a definition list?

See [Configure the Defaults ConfigMap](/serving/configuration/config-defaults.md).

TODO: Editing ConfigMaps guidance

> ### Autoscaling, high-availability and load balancing

Reviewer 2: Review capitalization of in-line references in this section.

Autoscaling includes [Configuring scale to zero](/serving/autoscaling/scale-to-zero.md), concurrency, [Configuring concurrency](/serving/autoscaling/concurrency.md), [Configuring the requests per second (RPS) target](/serving/autoscaling/rps-target.md), [Configuring scale bounds](/serving/autoscaling/scale-bounds.md), and [Additional autoscaling configuration for Knative Pod Autoscaler](/serving/autoscaling/kpa-specific.md)

See [Configuring high-availability components](/serving/config-ha.md).

For load balancing, you can [Configure Activator capacity](/serving/load-balancing/activator-capacity.md) and [Target-Burst capacity](/serving/load-balancing/target-burst-capacity.md).

### Networking

Networking administrative tasks include [Exclude namespaces from the Knative webhook](/serving/webhook-customizations.md), [Configuring the ingress gateway](/serving/setting-up-custom-ingress-gateway.md), [Configuring domain names](/serving/using-a-custom-domain.md), and [Feature and extension flags](/serving/configuration/feature-flags.md).

### Encryption

> You can have apply encryption as needed to your domain and clusters:

Reviewer 2: ```suggestion
You can apply encryption as needed to your domain and clusters:
```

### Eventing

The configuration options for eventing comprise the following areas:

- Event sources, see [Configure event source defaults](/eventing/configuration/sources-configuration.md) and also refer to [Knative reference mapping](/eventing/features/kreference-mapping.md).
- Brokers. See [Developer configuration options](/eventing/brokers/broker-developer-config-options.md) and [Configure Broker defaults](/eventing/configuration/broker-configuration.md) for an overview of broker configurations and an example.
- Kafka, a distributed event store and stream-processing platform. See [Configure Kafka features](/eventing/brokers/broker-types/kafka-broker/configuring-kafka-features.md) and [Configure Apache Kafka Channel defaults](/eventing/configuration/kafka-channel-configuration.md). See also [Configure KEDA Autoscaling of Knative Kafka Resources](/eventing/configuration/keda-configuration.md)
- Istio, a programmable, application-aware network. See [Eventing with Istio](/eventing/features/istio-integration.md).
- Channels. See [Configure Channel defaults](/eventing/configuration/channel-configuration.md).

> 

Reviewer 2: This would be more readable in a definition list.

## Monitoring and Observability

Knative is instrumented with OpenTelemetry. For metrics, see the [Metrics Reference](/serving/observability/metrics/serving-metrics.md) and [Collecting metrics](/serving/observability/metrics/collecting-metrics.md). You can access [CloudEvents traces](/serving/observability/accessing-traces.md).

For logging Serving activity, see [Collecting logs](/serving/observability/logging/collecting-logs.md), [Configuring logging](/serving/observability/logging/config-logging.md), and [Configuring Request logging](/serving/observability/logging/request-logging.md).

For logging Eventing activity, see [Collecting logs](/eventing/observability/logging/collecting-logs.md) and [Configuring logging](/eventing/observability/logging/config-logging.md).

### Blog articles

| Article | Description |
| --- | --- |
| [Monitoring Virtual Machines with Knative Eventing](../../blog/articles/kubevirt_meets_eventing.md) | Monitor KubeVirt VMs with Knative Eventing building blocks. |
| [Distributed tracing with Knative, OpenTelemetry and Jaeger](../../blog/articles/distributed-tracing.md) | Describes using Jaeger through a mix of OpenCensus and OpenTelemetry. |

> ## Security and Access Control

Reviewer 2: ```suggestion
Enforcing security
```

> [Security-Guard](/serving/app-security/security-guard-about.md) provides visibility into the security status of applications deployed by Knative Services. See [Verifying Knative Images](/reference/security/verifying-images.md) for verifying binaries for authenticity before deploying.

Reviewer 2: ```suggestion
[Security-Guard](/serving/app-security/security-guard-about.md) provides visibility into the security status of applications deployed by Knative Services. See [Verifying Knative Images](/reference/security/verifying-images.md) for verifying authenticity of binaries.
```

For access control, see [Transport Encryption](/eventing/features/transport-encryption.md), [Sender Identity](/eventing/features/sender-identity.md), and [cert-manager](/install/installing-cert-manager.md).

> ### Blog articles

Reviewer 2: See above re blogs.

| Article | Description |
| --- | --- |
| [Knative and Backstage](../../blog/articles/knative-backstage-plugins.md) | A platform for building developer portals.|
| [Knative Backstage Security](../../blog/articles/knative-backstage-security.md) | How to secure the communication between the Knative Event Mesh plugin and the backend. |

> ## Updates and Maintenance

Reviewer 2: ```suggestion
Updating and maintaining
```

> For Knative updates, see [About upgrading Knative](/install/upgrade/README.md) and [Checking your Knative version](/install/upgrade/check-install-version.md). The current Knative version is the shown in the version drop-down in the upper-left corner of this page. Refer to  [Upgrading with the Knative Operator](/install/upgrade/upgrade-installation-with-operator.md) or [Upgrading with kubectl](/install/upgrade/upgrade-installation.md).

Reviewer 2: Change
"About upgrading Knative"
to
"Upgrading Knative"
(here and the actual linked heading).

---
audience: administrator
components:
  - serving
  - eventing
function: reference
---
# Overview

> This page provides guidance for administrators on how to install and manage Knative on an existing Kubernetes cluster, and assumes you have familiarity the following:

Reviewer 2: ```suggestion
This page provides guidance for administrators on how to install and manage Knative on an existing Kubernetes cluster. It assumes you are familiar with the following:

- Kubernetes and Kubernetes administration.

> - The `kubectl`CLI tool. You can use existing Kubernetes management tools (policy, quota, etc) to manage Knative workloads.

Reviewer 2: ``` suggestion
- The `kubectl` CLI tool. You can use existing Kubernetes management tools (policy, quota, etc) to manage Knative workloads.
```

> - The Cloud Native Computing Foundation (CNCF) for which Knative is one of its projects, along with Kubernetes, Prometheus, and Istio.

Reviewer 2: ```suggestion
- The Cloud Native Computing Foundation (CNCF) an advocacy and support organization for open-source cloud software projects including Knative, Kubernetes, Prometheus, and Istio.
```

Why does the user need to be familiar with CNCF? Does that affect the administration tasks in any way? If not, I'd remove this requirement. CNCF ownership of Knative should be discussed elsewhere, for example in governance, support, and communications.

If this does need to be included here, add a link to the CNCF website and/or to the CNCF project pages for these projects.

> Additionally, you should have cluster-admin permissions or equivalent to to install software and manage resources in all clusters in the namespace.

Reviewer 2: Why "additionally"? Just list this as one of the requirement bullet items.

To simplify Knative installation and administration, you can use the Knative operator and the Knative CLI tool, but they are not required.

> Essentially, Knative aims to extend Kubernetes, and build on existing capabilities where feasible. It has two main underlying components that support plugging in multiple underlying transports within the same cluster:

Reviewer 2: ```suggestion
Knative aims to extend Kubernetes, and builds on existing capabilities where feasible. It has two main underlying components that support plugging in multiple underlying transports within the same cluster:
```

- Serving: Pods and pluggable network ingress routes.

> - Eventing: Pods and pluggable message transports (e.g. Kafka, RabbitMQ)

Reviewer 2: Use definition lists?

> Knative has default lightweight implementations if you don't already have a solution.

Reviewer 2: ```suggestion
Knative has a default lightweight messaging implementation if you don't already have a solution.
```

> Knative has default lightweight implementations if you don't already have a solution.

Reviewer: I feel like we need a transition sentence that explains something about managing the lifecycle of Knative, or maybe:

> As a cluster administrator, you should not need to directly manage applications and their supporting network and messaging resources on a day-to-day basis; Knative enabled application developers to manage their own functions and configure networking and message infrastructure within the bounds of their own namespace.  Cluster administrators are needed for the following operations:
>
> * Installing and upgrading the Knative components on the cluster
> * Creating Kubernetes namespaces for development teams, and managing controls on these namespaces
> * Configuring Knative and supporting infrastructure like message brokers and DNS
> * Securing the underlying Kubernetes cluster and Knative installation
>
> The administration documentation is intended to help you accomplish these tasks.

## Installing Knative

> You install Knative using YAML files and other resources either aided or not by the Knative Operator. The Knative Operator is a custom controller that extends the Kubernetes API to install Knative components. It allows you to automate applying the content, along with patching the contents to customize them. You install the Knative Operator either by using the Knative CLI Operator Plugin or by using KS8 Manifests or by Yelm.

Reviewer 2: ```suggestion
You install Knative using YAML files and other resources either aided or not by the Knative Operator. The Knative Operator is a custom controller that extends the Kubernetes API to install Knative components. It enables you to automate applying the content and patching the contents to customize them. You install the Knative Operator either by using the Knative CLI Operator Plugin, or by using KS8 Manifests or Yelm.

Here are the considerations for installing using YAML or the Knative Operator:

> | YAML-based install | Knative Operator install|

Reviewer: ```suggestion
| YAML installation | Knative Operator |
```

- Install the [Knative Operator](/install/operator/knative-with-operators.md) using Manifests or Yelm, and then use `kubectl` to install Knative components.

    This option alleviates complexity by using the Knative Operator, while still enabling purpose-built manageability using popular tools. It also gives you a separation of the core Knative application definition and the ConfigMap and other changes you make.

- Install the [Knative Operator CLI plugin](/install/operator/knative-with-operator-cli.md) and install the Knative Operator and the Knative CLI `kn` to  Knative components.

    This is the easiest install option and suitable for using if customization is not a concern.

The following table summarizes the options.

| Install option | Resources | kubectl CLI | kn CLI |
| --- | --- | --- | --- |
| YAML-based | All YAML prepared | Install components | not used |
| Knative Operator | Install Knative Operator using Manifests or Yelm |Install components | not used |

> | Knative Operator | Install Knative Operator CLI plugin | not used | Install components |

Reviewer: This table is kind of weird, particularly since the previous table was column-major, but this one is row-major.

Reviewer: I'd probably drop this table, though the other table seemed informative.

Knative supports subsequent installs after the initial installation, you so your initial choices don't lock you in. For example, you can migrate from one message transport or network ingress to another without losing messages.

> ## Defining and modifying custom resources

Reviewer: It's not clear to me if this is supposed to be "Configuring Knative" or something else.  If it is "Configuring Knative", I'd suggest an intro like the following (this one is kinda bad, and needs a rewrite, but getting notes out quickly):

> A Knative installation consists of several custom resources in the `knative.dev` API namespace, and controllers running on the cluster which manage those custom resources.  During installation and while operating the cluster, you may need to customize the behavior of the controller.  Knative controllers read their configuration from ConfigMaps in the cluster, and can generally respond to changes in the ConfigMaps live, without needing a reload.  In some cases, it may be necessary to restart the controller after updating a ConfigMap; this should be well-documented by the documentation in the ConfigMap.
>
> Knative installs ConfigMaps for each tracked configuration file, along with validation of the values in the configuration.  The ConfigMaps installed by Knative leave all the controller settings at their default values, but include an `_example` key which documents all the known keys and their operation as an entry within the ConfigMap.  **Note that if you perform an upgrade of a YAML installation, this will include new ConfigMaps which may overwrite your existing settings unless you preserve them.**
>
> Knative uses this configuration pattern (existing ConfigMaps with empty defaults and an `_example` key) to balance several different needs:
> * Deploying all the ConfigMaps to the cluster during the installation makes it easy for new administrators to find the relevant ConfigMaps
> * Not setting configuration keys in the ConfigMaps by default makes it easy for clusters to pick up new behavior and features during upgrade unless the feature is specifically intended to be held back.
> * The `_example` key allows the documentation of the configuration keys to be persisted on the Kubernetes apiserver, rather being lost (as YAML comments are) when applying the configuration to the server.
> * There is a webhook which will error if the `_example` key content is changed -- we have found that people often edit the example text, rather than setting a configuration key.

It may be worth providing an right/wrong example here:

```yaml
kind: ConfigMap
apiGroup: v1
metadata:
  name: config-example
data:
  _example: |
    # sample-key can be "enabled" or "disabled"
    sample-key: enabled
```

Right:
```yaml
kind: ConfigMap
apiGroup: v1
metadata:
  name: config-example
data:
  _example: |
    # sample-key can be "enabled" or "disabled"
    sample-key: enabled
  sample-key: disabled
```

wrong:

```yaml
kind: ConfigMap
apiGroup: v1
metadata:
  name: config-example
data:
  _example: |
    # sample-key can be "enabled" or "disabled"
    sample-key: disabled
```

You'll get a warning when trying to save the second one.  I don't know where you want to put that, but it's probably worth documenting somewhere on this page.

> Either before or after the installing Knative Eventing and Serving components, you can create and modify custom resources and reinstall components as needed. You do so by creating or modifying a ConfigMap using a custom resource definition (CRD).

Reviewer: This sentence is weird and not technically correct.  Folks should not be modifying the CRDs (which define the shapes of Knative resources), but should instead be editing the ConfigMaps in the controller namespace.  ConfigMap is a native Kubernetes type, which is basically a map from string-to-string.

> You customize resources using `kubectl` using the Knative Operator using `kn`. See [Knative Serving CRDs](/install/operator/configuring-serving-cr.md) and [Knative Eventing CRDs](/install/operator/configuring-eventing-cr.md).

Reviewer: This sentence is weird, because it talks about both `kubectl` and `kn`, but it's not clear which tool is doing what.

The following table lists the names of CRDs (metadata name) for the Serving and Eventing components. They are defined by `eventing-crds.yaml` and `serving-crds.yaml` in the [Knative Eventing installation files](/install/yaml-install/eventing/eventing-installation-files.md) and [Knative Serving installation Files](/install/yaml-install/serving/serving-installation-files.md), respectively.

| Eventing CRDs | Serving CRDs |
| --- | --- |

> | brokers.eventing.knative.dev<br>channels.messaging.knative.dev<br>eventpolicies.eventing.knative.dev<br>eventtransforms.eventing.knative.dev<br>eventtypes.eventing.knative.dev<br>integrationsinks.sinks.knative.dev<br>jobsinks.sinks.knative.dev<br>parallels.flows.knative.dev<br>requestreplies.eventing.knative.dev<br>sequences.flows.knative.dev<br>subscriptions.messaging.knative.dev<br>triggers.eventing.knative.dev | certificates.networking.internal.knative.dev<br>configurations.serving.knative.dev<br>clusterdomainclaims.networking.internal.knative.dev<br>domainmappings.serving.knative.dev<br>ingresses.networking.internal.knative.dev<br>metrics.autoscaling.internal.knative.dev<br>podautoscalers.autoscaling.internal.knative.dev<br>revisions.serving.knative.dev<br>routes.serving.knative.dev<br>serverlessservices.networking.internal.knative.dev<br>services.serving.knative.dev<br>images.caching.internal.knative.dev |

Reviewer: Folks can get these from a live cluster using `kubectl api-resources`, which is probably a better way to do so if needed.  I'm not sure this table adds much, so I'd probably remove it, again.

### Recommended plugins

You can also install these plugins service to extend Knative capabilities for service meshes:

- [Istio for Knative](/install/installing-istio.md) - Extends Kubernetes with a programmable, application-aware network.
- [Knative Backstage plugin](/install/installing-backstage-plugins.md) - An Event Mesh that allows you to view and manage Knative Eventing resources.

> - [cert-manager](/install/installing-cert-manager.md) - Provision and manage TLS certificates in Kubernetes.

Reviewer: It's weird to highlight these particular plugins.  It feels like it would be worth a diagram here explaining the pluggable components (the duck analogy I gave earlier, for example), and then giving some examples of the different ways things can plug in.  These three have three different plugin modes, so it's not totally obvious which are special cases:

### Networking plugins

Used by Serving to route incoming traffic.  Main plugins:
* Kourier (internal no-dependency option)
* Istio (service mesh)
* Courier (general-purpose ingress)
* Gateway API (beta)

### Messaging plugins:

* In-memory (internal no-dependency option)
* Kafka (in-order, high-thoughput but moderate complexity)
* RabbitMQ (configurable order, moderate throughput and complexity)
* NATS (low complexity)

### Optional plugins (can add or not, but they aren't fungible with each other)

* Cert-manager provides TLS certificate provisioning.  This is needed (and not swappable) if you want to secure ingress traffic.
* Backstage provides a UI for developers
* Prometheus provides metrics observability
* Jaeger provides tracing observability

## Administration tasks

> The following table lists configurations, extensibility, conversions, and other actions for Knative Administrators organized by technical area. Tasks are linked to procedures and other guidance.

Reviewer: We should just move these pages into the nav for administration, rather than building a big map in the file.  (This also makes the navigation easier, rather than needing to go back and forward, and may help with reorganizing things to be more logically grouped to administrators.)

## Monitoring and Observability

Both the Serving and Eventing components are well maintained with logging, tracing, and metric collections.

| Concepts and Procedures | Reference |
| --- | --- |
| Request traces<br>Collecting logs<br>Configuring logging<br>Configuring Request logging<br>Collecting metrics<br>Accessing CloudEvent traces<br>Collecting logs<br>Configuring logging<br>Collecting metrics | Service metrics<br>Metrics Reference |

## Security

Knative provides robust security and access control measures with the certificate manager, networking TLS certificates, domain and cluster encryption, and application security.

| Concepts and Procedures | Reference |
| --- | --- |
| Install cert-manager <br>Using a custom TLS certificate for DomainMapping<br>Configure cert-manager integration<br>Configure external domain encryptio><br>Configure cluster-local domain encryption<br>Configure Knative system-internal encryption<br>Installing Security-Guard<br>Security-guard quickstart<br>Security-Guard example alerts<br>Transport Encryption<br>Sender Identity<br>Verifying Knative Images | Using extensions enabled by QPOptions<br>About Security-Guard<br> |

## Updates and Maintenance

> In addition to updates, Knative provides several configurations for maintaining your clusters in optimal condition including resources, volume, traffic, and load balancing.

Reviewer: Again, I'd emphasize integrating with existing components for things like resource limits and volumes (using the existing Kubernetes feature).

The "gradual rollout of traffic" and "developer configuration options" are probably more appropriate for the developer documentation (most of the site) than cluster administrators.  There *are* settings administrators can use to adjust the default behavior for users that don't specify behaviors, so you might have a setting on "adjusting cluster defaults".  That generally feels like it falls under "configuration", though.
